src/main/scala/org/constellation/primitives/EdgeDAO.scala

```diff
@@ -378,8 +380,7 @@ class ThreadSafeSnapshotService(concurrentTipService: ConcurrentTipService) {
           }
           _ <- cb.storeSOE()
           maybeHeight <- calculateHeight(checkpoint)
-          metricKey = if (maybeHeight.isEmpty) Metrics.heightEmpty else Metrics.heightNonEmpty
-          _ <- dao.metrics.incrementMetricAsync(metricKey)
+          _ <- if(maybeHeight.isEmpty) dao.metrics.incrementMetricAsync(Metrics.heightEmpty).flatMap(_ => IO.raiseError(MissingHeightException(cb))) else IO.unit
```

Comment:
IMO raiseError is the main effect you want to trigger, thus:
`IO.raiseError(MissingHeightException(cb).flatTap(_ => dao.metrics.incrementMetricAsync(Metrics.heightEmpty)`

Reply:
error will be swallowed and not propagated to upper levels